@civanch while testing it. 

https://github.com/cms-sw/cmsdist/pull/2323

it results in error:

```
CMake Error at cmake/Modules/Geant4MacroLibraryTargets.cmake:109
(add_library):
   Cannot find source file:

/build/cmsbuild/jenkins-workarea/workspace/cmsdist-test-pr/testBuildDir/BUILD/slc6_amd64_gcc530/external/geant4/10.02.p01-cms/geant4.10.02.p01/source/physics_lists/constructors/hadron_inelastic/src/G4HadronPhysicsFTFP_BERT_ATL.cc

   Tried extensions .c .C .c++ .cc .cpp .cxx .m .M .mm .h .hh .h++ .hm .hpp
   .hxx .in .txx
```

I think there is some file missing from the commit.

--- a/source/physics_lists/constructors/hadron_inelastic/include/G4HadronPhysicsFTFP_BERT_ATL.hh
+++ b/source/physics_lists/constructors/hadron_inelastic/include/G4HadronPhysicsFTFP_BERT_ATL.hh
@@ -1,0 +1,109 @@
+//
+// ********************************************************************
+// * License and Disclaimer                                           *
+// *                                                                  *
+// * The  Geant4 software  is  copyright of the Copyright Holders  of *
+// * the Geant4 Collaboration.  It is provided  under  the terms  and *
+// * conditions of the Geant4 Software License,  included in the file *
+// * LICENSE and available at  http://cern.ch/geant4/license .  These *
+// * include a list of copyright holders.                             *
+// *                                                                  *
+// * Neither the authors of this software system, nor their employing *
+// * institutes,nor the agencies providing financial support for this *
+// * work  make  any representation or  warranty, express or implied, *
+// * regarding  this  software system or assume any liability for its *
+// * use.  Please see the license in the file  LICENSE  and URL above *
+// * for the full disclaimer and the limitation of liability.         *
+// *                                                                  *
+// * This  code  implementation is the result of  the  scientific and *
+// * technical work of the GEANT4 collaboration.                      *
+// * By using,  copying,  modifying or  distributing the software (or *
+// * any work based  on the software)  you  agree  to acknowledge its *
+// * use  in  resulting  scientific  publications,  and indicate your *
+// * acceptance of all terms of the Geant4 Software license.          *
+// ********************************************************************
+//
+// $Id:$
+//
+//---------------------------------------------------------------------------
+// Author: Alberto Ribon
+// Date:   April 2016
+//
+// Hadron physics for the new physics list FTFP_BERT_ATL.
+// This is a modified version of the FTFP_BERT hadron physics for ATLAS.
+// The hadron physics of FTFP_BERT_ATL has the transition between Bertini
+// (BERT) intra-nuclear cascade model and Fritiof (FTF) string model in the
+// energy region [9, 12] GeV (instead of [4, 5] GeV as in FTFP_BERT).
+//---------------------------------------------------------------------------
+//
+#ifndef G4HadronPhysicsFTFP_BERT_ATL_h
+#define G4HadronPhysicsFTFP_BERT_ATL_h 1
+
+#include "globals.hh"
+#include "G4ios.hh"
+
+#include "G4VPhysicsConstructor.hh"
+
+#include "G4PiKBuilder.hh"
+#include "G4BertiniPiKBuilder.hh"
+#include "G4FTFPPiKBuilder.hh"
+
+#include "G4ProtonBuilder.hh"
+#include "G4BertiniProtonBuilder.hh"
+#include "G4FTFPNeutronBuilder.hh"
+#include "G4FTFPProtonBuilder.hh"
+
+#include "G4NeutronBuilder.hh"
+#include "G4BertiniNeutronBuilder.hh"
+#include "G4FTFPNeutronBuilder.hh"
+
+#include "G4HyperonFTFPBuilder.hh"
+#include "G4AntiBarionBuilder.hh"
+#include "G4FTFPAntiBarionBuilder.hh"
+
+class G4ComponentGGHadronNucleusXsc;
+
+
+class G4HadronPhysicsFTFP_BERT_ATL : public G4VPhysicsConstructor
+{
+  public: 
+    G4HadronPhysicsFTFP_BERT_ATL(G4int verbose =1);
+    G4HadronPhysicsFTFP_BERT_ATL(const G4String& name, G4bool quasiElastic=false);
+    virtual ~G4HadronPhysicsFTFP_BERT_ATL();
+
+  public: 
+    virtual void ConstructParticle();
+    virtual void ConstructProcess();
+
+  private:
+    void CreateModels();
+    G4bool QuasiElastic;
+
+    // Simplify handling of TLS data, encapsulate everyhing in a structure
+    struct ThreadPrivate { 
+      G4NeutronBuilder * theNeutrons;
+      G4BertiniNeutronBuilder * theBertiniNeutron;
+      G4FTFPNeutronBuilder * theFTFPNeutron;
+ 
+      G4PiKBuilder * thePiK;
+      G4BertiniPiKBuilder * theBertiniPiK;
+      G4FTFPPiKBuilder * theFTFPPiK;
+    
+      G4ProtonBuilder * thePro;
+      G4BertiniProtonBuilder * theBertiniPro;
+      G4FTFPProtonBuilder * theFTFPPro;    
+    
+      G4HyperonFTFPBuilder * theHyperon;
+    
+      G4AntiBarionBuilder * theAntiBaryon;
+      G4FTFPAntiBarionBuilder * theFTFPAntiBaryon;
+
+      G4ComponentGGHadronNucleusXsc * xsKaon;
+      G4VCrossSectionDataSet * xsNeutronInelasticXS;
+      G4VCrossSectionDataSet * xsNeutronCaptureXS;
+    };
+    static G4ThreadLocal ThreadPrivate* tpdata;
+};
+
+#endif
+

--- a/source/physics_lists/constructors/hadron_inelastic/src/G4HadronPhysicsFTFP_BERT_ATL.cc
+++ b/source/physics_lists/constructors/hadron_inelastic/src/G4HadronPhysicsFTFP_BERT_ATL.cc
@@ -1,0 +1,222 @@
+//
+// ********************************************************************
+// * License and Disclaimer                                           *
+// *                                                                  *
+// * The  Geant4 software  is  copyright of the Copyright Holders  of *
+// * the Geant4 Collaboration.  It is provided  under  the terms  and *
+// * conditions of the Geant4 Software License,  included in the file *
+// * LICENSE and available at  http://cern.ch/geant4/license .  These *
+// * include a list of copyright holders.                             *
+// *                                                                  *
+// * Neither the authors of this software system, nor their employing *
+// * institutes,nor the agencies providing financial support for this *
+// * work  make  any representation or  warranty, express or implied, *
+// * regarding  this  software system or assume any liability for its *
+// * use.  Please see the license in the file  LICENSE  and URL above *
+// * for the full disclaimer and the limitation of liability.         *
+// *                                                                  *
+// * This  code  implementation is the result of  the  scientific and *
+// * technical work of the GEANT4 collaboration.                      *
+// * By using,  copying,  modifying or  distributing the software (or *
+// * any work based  on the software)  you  agree  to acknowledge its *
+// * use  in  resulting  scientific  publications,  and indicate your *
+// * acceptance of all terms of the Geant4 Software license.          *
+// ********************************************************************
+//
+// $Id:$
+//
+//---------------------------------------------------------------------------
+// Author: Alberto Ribon
+// Date:   April 2016
+//
+// Hadron physics for the new physics list FTFP_BERT_ATL.
+// This is a modified version of the FTFP_BERT hadron physics for ATLAS.
+// The hadron physics of FTFP_BERT_ATL has the transition between Bertini
+// (BERT) intra-nuclear cascade model and Fritiof (FTF) string model in the
+// energy region [9, 12] GeV (instead of [4, 5] GeV as in FTFP_BERT).
+//---------------------------------------------------------------------------
+//
+#include <iomanip>   
+
+#include "G4HadronPhysicsFTFP_BERT_ATL.hh"
+
+#include "globals.hh"
+#include "G4ios.hh"
+#include "G4SystemOfUnits.hh"
+#include "G4ParticleDefinition.hh"
+#include "G4ParticleTable.hh"
+
+#include "G4MesonConstructor.hh"
+#include "G4BaryonConstructor.hh"
+#include "G4ShortLivedConstructor.hh"
+
+#include "G4ComponentGGHadronNucleusXsc.hh"
+#include "G4CrossSectionInelastic.hh"
+#include "G4HadronCaptureProcess.hh"
+#include "G4NeutronRadCapture.hh"
+#include "G4NeutronInelasticXS.hh"
+#include "G4NeutronCaptureXS.hh"
+
+#include "G4CrossSectionDataSetRegistry.hh"
+
+#include "G4PhysListUtil.hh"
+
+// factory
+#include "G4PhysicsConstructorFactory.hh"
+//
+G4_DECLARE_PHYSCONSTR_FACTORY(G4HadronPhysicsFTFP_BERT_ATL);
+
+G4ThreadLocal G4HadronPhysicsFTFP_BERT_ATL::ThreadPrivate* G4HadronPhysicsFTFP_BERT_ATL::tpdata=0;
+
+G4HadronPhysicsFTFP_BERT_ATL::G4HadronPhysicsFTFP_BERT_ATL(G4int)
+    :  G4VPhysicsConstructor("hInelastic FTFP_BERT_ATL")
+/*    , theNeutrons(0)
+    , theBertiniNeutron(0)
+    , theFTFPNeutron(0)
+    , thePiK(0)
+    , theBertiniPiK(0)
+    , theFTFPPiK(0)
+    , thePro(0)
+    , theBertiniPro(0)
+    , theFTFPPro(0)
+    , theHyperon(0)
+    , theAntiBaryon(0)
+    , theFTFPAntiBaryon(0) */
+    , QuasiElastic(false)
+  /*, xsKaon(0)
+    , xsNeutronInelasticXS(0)
+    , xsNeutronCaptureXS(0)*/
+{}
+
+G4HadronPhysicsFTFP_BERT_ATL::G4HadronPhysicsFTFP_BERT_ATL(const G4String& name, G4bool quasiElastic)
+    :  G4VPhysicsConstructor(name) 
+/*    , theNeutrons(0)
+    , theBertiniNeutron(0)
+    , theFTFPNeutron(0)
+    , thePiK(0)
+    , theBertiniPiK(0)
+    , theFTFPPiK(0)
+    , thePro(0)
+    , theBertiniPro(0)
+    , theFTFPPro(0)
+    , theHyperon(0)
+    , theAntiBaryon(0)
+    , theFTFPAntiBaryon(0)*/
+    , QuasiElastic(quasiElastic)
+  /*, xsKaonMinus(0)
+    , xsNeutronInelasticXS(0)
+    , xsNeutronCaptureXS(0)*/
+{}
+
+void G4HadronPhysicsFTFP_BERT_ATL::CreateModels()
+{
+  G4double minFTFP =  9.0 * GeV;
+  G4double maxBERT = 12.0 * GeV;
+  G4cout << " FTFP_BERT_ATL : new threshold between BERT and FTFP" 
+         << " is over the interval " << minFTFP/GeV << " to " << maxBERT/GeV 
+         << " GeV." << G4endl;
+  QuasiElastic= false; 
+
+  tpdata->theNeutrons=new G4NeutronBuilder;
+  tpdata->theNeutrons->RegisterMe(tpdata->theFTFPNeutron=new G4FTFPNeutronBuilder(QuasiElastic));
+  tpdata->theFTFPNeutron->SetMinEnergy(minFTFP);
+  tpdata->theNeutrons->RegisterMe(tpdata->theBertiniNeutron=new G4BertiniNeutronBuilder);
+  tpdata->theBertiniNeutron->SetMinEnergy(0.0*GeV);
+  tpdata->theBertiniNeutron->SetMaxEnergy(maxBERT);
+
+  tpdata->thePro=new G4ProtonBuilder;
+  tpdata->thePro->RegisterMe(tpdata->theFTFPPro=new G4FTFPProtonBuilder(QuasiElastic));
+  tpdata->theFTFPPro->SetMinEnergy(minFTFP);
+  tpdata->thePro->RegisterMe(tpdata->theBertiniPro=new G4BertiniProtonBuilder);
+  tpdata->theBertiniPro->SetMaxEnergy(maxBERT);
+
+  tpdata->thePiK=new G4PiKBuilder;
+  tpdata->thePiK->RegisterMe(tpdata->theFTFPPiK=new G4FTFPPiKBuilder(QuasiElastic));
+  tpdata->theFTFPPiK->SetMinEnergy(minFTFP);
+  tpdata->thePiK->RegisterMe(tpdata->theBertiniPiK=new G4BertiniPiKBuilder);
+  tpdata->theBertiniPiK->SetMaxEnergy(maxBERT);
+  
+  tpdata->theHyperon=new G4HyperonFTFPBuilder;
+    
+  tpdata->theAntiBaryon=new G4AntiBarionBuilder;
+  tpdata->theAntiBaryon->RegisterMe(tpdata->theFTFPAntiBaryon=new G4FTFPAntiBarionBuilder(QuasiElastic));
+}
+
+G4HadronPhysicsFTFP_BERT_ATL::~G4HadronPhysicsFTFP_BERT_ATL()
+{
+  if (!tpdata) return;
+
+  delete tpdata->theNeutrons;
+  delete tpdata->theBertiniNeutron;
+  delete tpdata->theFTFPNeutron;
+
+  delete tpdata->thePiK;
+  delete tpdata->theBertiniPiK;
+  delete tpdata->theFTFPPiK;
+    
+  delete tpdata->thePro;
+  delete tpdata->theBertiniPro;
+  delete tpdata->theFTFPPro;    
+    
+  delete tpdata->theHyperon;
+  delete tpdata->theAntiBaryon;
+  delete tpdata->theFTFPAntiBaryon;
+ 
+  //Note that here we need to set to 0 the pointer
+  //since tpdata is static and if thread are "reused"
+  //it can be problematic
+  delete tpdata; tpdata = 0;
+}
+
+void G4HadronPhysicsFTFP_BERT_ATL::ConstructParticle()
+{
+  G4MesonConstructor pMesonConstructor;
+  pMesonConstructor.ConstructParticle();
+
+  G4BaryonConstructor pBaryonConstructor;
+  pBaryonConstructor.ConstructParticle();
+
+  G4ShortLivedConstructor pShortLivedConstructor;
+  pShortLivedConstructor.ConstructParticle();  
+}
+
+#include "G4ProcessManager.hh"
+void G4HadronPhysicsFTFP_BERT_ATL::ConstructProcess()
+{
+  if ( tpdata == 0 ) tpdata = new ThreadPrivate;
+  CreateModels();
+  tpdata->theNeutrons->Build();
+  tpdata->thePro->Build();
+  tpdata->thePiK->Build();
+
+  // --- Kaons ---
+  tpdata->xsKaon = new G4ComponentGGHadronNucleusXsc();
+  G4VCrossSectionDataSet * kaonxs = new G4CrossSectionInelastic(tpdata->xsKaon);
+  G4PhysListUtil::FindInelasticProcess(G4KaonMinus::KaonMinus())->AddDataSet(kaonxs);
+  G4PhysListUtil::FindInelasticProcess(G4KaonPlus::KaonPlus())->AddDataSet(kaonxs);
+  G4PhysListUtil::FindInelasticProcess(G4KaonZeroShort::KaonZeroShort())->AddDataSet(kaonxs);
+  G4PhysListUtil::FindInelasticProcess(G4KaonZeroLong::KaonZeroLong())->AddDataSet(kaonxs);
+    
+  tpdata->theHyperon->Build();
+  tpdata->theAntiBaryon->Build();
+
+  // --- Neutrons ---
+  tpdata->xsNeutronInelasticXS = (G4NeutronInelasticXS*)G4CrossSectionDataSetRegistry::Instance()->GetCrossSectionDataSet(G4NeutronInelasticXS::Default_Name());
+  G4PhysListUtil::FindInelasticProcess(G4Neutron::Neutron())->AddDataSet(tpdata->xsNeutronInelasticXS);
+
+  G4HadronicProcess* capture = 0;
+  G4ProcessManager* pmanager = G4Neutron::Neutron()->GetProcessManager();
+  G4ProcessVector*  pv = pmanager->GetProcessList();
+  for ( size_t i=0; i < static_cast<size_t>(pv->size()); ++i ) {
+    if ( fCapture == ((*pv)[i])->GetProcessSubType() ) {
+      capture = static_cast<G4HadronicProcess*>((*pv)[i]);
+    }
+  }
+  if ( ! capture ) {
+    capture = new G4HadronCaptureProcess("nCapture");
+    pmanager->AddDiscreteProcess(capture);
+  }
+  tpdata->xsNeutronCaptureXS = (G4NeutronCaptureXS*)G4CrossSectionDataSetRegistry::Instance()->GetCrossSectionDataSet(G4NeutronCaptureXS::Default_Name());
+  capture->AddDataSet(tpdata->xsNeutronCaptureXS);
+  capture->RegisterMe(new G4NeutronRadCapture());
+}

--- a/source/physics_lists/lists/include/FTFP_BERT_ATL.hh
+++ b/source/physics_lists/lists/include/FTFP_BERT_ATL.hh
@@ -1,0 +1,66 @@
+//
+// ********************************************************************
+// * License and Disclaimer                                           *
+// *                                                                  *
+// * The  Geant4 software  is  copyright of the Copyright Holders  of *
+// * the Geant4 Collaboration.  It is provided  under  the terms  and *
+// * conditions of the Geant4 Software License,  included in the file *
+// * LICENSE and available at  http://cern.ch/geant4/license .  These *
+// * include a list of copyright holders.                             *
+// *                                                                  *
+// * Neither the authors of this software system, nor their employing *
+// * institutes,nor the agencies providing financial support for this *
+// * work  make  any representation or  warranty, express or implied, *
+// * regarding  this  software system or assume any liability for its *
+// * use.  Please see the license in the file  LICENSE  and URL above *
+// * for the full disclaimer and the limitation of liability.         *
+// *                                                                  *
+// * This  code  implementation is the result of  the  scientific and *
+// * technical work of the GEANT4 collaboration.                      *
+// * By using,  copying,  modifying or  distributing the software (or *
+// * any work based  on the software)  you  agree  to acknowledge its *
+// * use  in  resulting  scientific  publications,  and indicate your *
+// * acceptance of all terms of the Geant4 Software license.          *
+// ********************************************************************
+//
+// $Id:$
+//
+//---------------------------------------------------------------------------
+// Author: Alberto Ribon
+// Date:   April 2016
+//
+// New physics list FTFP_BERT_ATL.
+// This is a modified version of the FTFP_BERT physics list for ATLAS.
+// The physics list FTFP_BERT_ATL has the transition between Bertini (BERT)
+// intra-nuclear cascade model and Fritiof (FTF) string model in the
+// energy region [9, 12] GeV (instead of [4, 5] GeV as in FTFP_BERT).
+//----------------------------------------------------------------------------
+//
+#ifndef TFTFP_BERT_ATL_h
+#define TFTFP_BERT_ATL_h 1
+
+#include <CLHEP/Units/SystemOfUnits.h>
+
+#include "globals.hh"
+#include "G4VModularPhysicsList.hh"
+#include "CompileTimeConstraints.hh"
+
+template<class T>
+class TFTFP_BERT_ATL: public T
+{
+public:
+  TFTFP_BERT_ATL(G4int ver = 1);
+  virtual ~TFTFP_BERT_ATL();
+  
+public:
+  // SetCuts() 
+  virtual void SetCuts();
+
+private:
+  enum {ok = CompileTimeConstraints::IsA<T, G4VModularPhysicsList>::ok };
+};
+#include "FTFP_BERT_ATL.icc"
+typedef TFTFP_BERT_ATL<G4VModularPhysicsList> FTFP_BERT_ATL;
+
+#endif
+

--- a/source/physics_lists/lists/include/FTFP_BERT_ATL.icc
+++ b/source/physics_lists/lists/include/FTFP_BERT_ATL.icc
@@ -1,0 +1,119 @@
+//
+// ********************************************************************
+// * License and Disclaimer                                           *
+// *                                                                  *
+// * The  Geant4 software  is  copyright of the Copyright Holders  of *
+// * the Geant4 Collaboration.  It is provided  under  the terms  and *
+// * conditions of the Geant4 Software License,  included in the file *
+// * LICENSE and available at  http://cern.ch/geant4/license .  These *
+// * include a list of copyright holders.                             *
+// *                                                                  *
+// * Neither the authors of this software system, nor their employing *
+// * institutes,nor the agencies providing financial support for this *
+// * work  make  any representation or  warranty, express or implied, *
+// * regarding  this  software system or assume any liability for its *
+// * use.  Please see the license in the file  LICENSE  and URL above *
+// * for the full disclaimer and the limitation of liability.         *
+// *                                                                  *
+// * This  code  implementation is the result of  the  scientific and *
+// * technical work of the GEANT4 collaboration.                      *
+// * By using,  copying,  modifying or  distributing the software (or *
+// * any work based  on the software)  you  agree  to acknowledge its *
+// * use  in  resulting  scientific  publications,  and indicate your *
+// * acceptance of all terms of the Geant4 Software license.          *
+// ********************************************************************
+//
+// $Id:$
+//
+//---------------------------------------------------------------------------
+// Author: Alberto Ribon
+// Date:   April 2016
+//
+// New physics list FTFP_BERT_ATL.
+// This is a modified version of the FTFP_BERT physics list for ATLAS.
+// The physics list FTFP_BERT_ATL has the transition between Bertini (BERT)
+// intra-nuclear cascade model and Fritiof (FTF) string model in the
+// energy region [9, 12] GeV (instead of [4, 5] GeV as in FTFP_BERT).
+//----------------------------------------------------------------------------
+//
+#include <iomanip>   
+
+#include "globals.hh"
+#include "G4ios.hh"
+#include "G4ProcessManager.hh"
+#include "G4ProcessVector.hh"
+#include "G4ParticleTypes.hh"
+#include "G4ParticleTable.hh"
+
+#include "G4Material.hh"
+#include "G4MaterialTable.hh"
+
+#include "G4DecayPhysics.hh"
+#include "G4EmStandardPhysics.hh"
+#include "G4EmExtraPhysics.hh"
+#include "G4IonPhysics.hh"
+#include "G4StoppingPhysics.hh"
+#include "G4HadronElasticPhysics.hh"
+#include "G4NeutronTrackingCut.hh"
+
+#include "G4DataQuestionaire.hh"
+#include "G4HadronPhysicsFTFP_BERT_ATL.hh"
+
+#include "G4WarnPLStatus.hh"
+
+
+template<class T> TFTFP_BERT_ATL<T>::TFTFP_BERT_ATL(G4int ver):  T()
+{
+  // default cut value  (1.0mm) 
+  // defaultCutValue = 1.0*CLHEP::mm;
+  G4DataQuestionaire it(photon);
+  G4cout << "<<< Geant4 Physics List simulation engine: FTFP_BERT_ATL 2.0"<<G4endl;
+  G4cout <<G4endl;
+  this->defaultCutValue = 0.7*CLHEP::mm;  
+  this->SetVerboseLevel(ver);
+
+  G4WarnPLStatus exp;
+  exp.Experimental("FTFP_BERT_ATL");
+
+  // EM Physics
+  this->RegisterPhysics( new G4EmStandardPhysics(ver));
+
+  // Synchroton Radiation & GN Physics
+  this->RegisterPhysics( new G4EmExtraPhysics(ver) );
+
+  // Decays 
+  this->RegisterPhysics( new G4DecayPhysics(ver) );
+
+   // Hadron Elastic scattering
+  this->RegisterPhysics( new G4HadronElasticPhysics(ver) );
+
+   // Hadron Physics
+  this->RegisterPhysics( new G4HadronPhysicsFTFP_BERT_ATL(ver) );
+
+  // Stopping Physics
+  this->RegisterPhysics( new G4StoppingPhysics(ver) );
+
+  // Ion Physics
+  this->RegisterPhysics( new G4IonPhysics(ver));
+  
+  // Neutron tracking cut
+  this->RegisterPhysics( new G4NeutronTrackingCut(ver));
+}
+
+template<class T> TFTFP_BERT_ATL<T>::~TFTFP_BERT_ATL()
+{
+}
+
+template<class T> void TFTFP_BERT_ATL<T>::SetCuts()
+{
+  if (this->verboseLevel >1){
+    G4cout << "FTFP_BERT_ATL::SetCuts:";
+  }  
+  //  " G4VUserPhysicsList::SetCutsWithDefault" method sets 
+  //   the default cut value for all particle types 
+
+  this->SetCutsWithDefault();   
+ 
+//  if (this->verboseLevel > 0)
+//    G4VUserPhysicsList::DumpCutValuesTable();  
+}

